File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/ICommentsView.java
package dev.ragnarok.fenrir.mvp.view;

import androidx.annotation.StringRes;

import java.util.List;

import dev.ragnarok.fenrir.model.Comment;
import dev.ragnarok.fenrir.model.LoadMoreState;
import dev.ragnarok.fenrir.model.Owner;
import dev.ragnarok.fenrir.model.Sticker;
import dev.ragnarok.fenrir.model.User;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface ICommentsView extends IAccountDependencyView, IAttachmentsPlacesView, IMvpView,
        IErrorView, IToolbarView, IProgressView, IToastView {

    void displayData(List<Comment> data);

    void notifyDataSetChanged();

    void setupLoadUpHeader(@LoadMoreState int state);

    void setupLoadDownFooter(@LoadMoreState int state);

    void notifyDataAddedToTop(int count);

    void notifyDataAddedToBottom(int count);

    void notifyItemChanged(int index);

    void moveFocusTo(int index, boolean smooth);

    void displayBody(String body);

    void displayAttachmentsCount(int count);

    void setButtonSendAvailable(boolean available);

    void openAttachmentsManager(int accountId, Integer draftCommentId, int sourceOwnerId, String draftCommentBody);

    void setupReplyViews(String replyTo);

    void replaceBodySelectionTextTo(String replyText);

    void goToCommentEdit(int accountId, Comment comment, Integer commemtId);

    void goToWallPost(int accountId, int postId, int postOwnerId);

    void goToVideoPreview(int accountId, int videoId, int videoOwnerId);

    void banUser(int accountId, int groupId, User user);

    void displayAuthorAvatar(String url);

    void showAuthorSelectDialog(List<Owner> owners);

    void scrollToPosition(int position);

    void showCommentSentToast();

    void setupOptionMenu(boolean topicPollAvailable, boolean gotoSourceAvailable, @StringRes Integer gotoSourceText);

    void setEpmtyTextVisible(boolean visible);

    void setCenterProgressVisible(boolean visible);

    void displayDeepLookingCommentProgress();

    void dismissDeepLookingCommentProgress();

    void setCanSendSelectAuthor(boolean can);

    void updateStickers(List<Sticker> items);

    interface ICommentContextView {
        void setCanEdit(boolean can);

        void setCanDelete(boolean can);

        void setCanBan(boolean can);
    }
}

File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/ICommentsView.kt
package dev.ragnarok.fenrir.mvp.view

import androidx.annotation.StringRes
import dev.ragnarok.fenrir.model.*
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface ICommentsView : IAccountDependencyView, IAttachmentsPlacesView, IMvpView, IErrorView,
    IToolbarView, IProgressView, IToastView {
    fun displayData(data: MutableList<Comment>)
    fun notifyDataSetChanged()
    fun setupLoadUpHeader(@LoadMoreState state: Int)
    fun setupLoadDownFooter(@LoadMoreState state: Int)
    fun notifyDataAddedToTop(count: Int)
    fun notifyDataAddedToBottom(count: Int)
    fun notifyItemChanged(index: Int)
    fun moveFocusTo(index: Int, smooth: Boolean)
    fun displayBody(body: String?)
    fun displayAttachmentsCount(count: Int)
    fun setButtonSendAvailable(available: Boolean)
    fun openAttachmentsManager(
        accountId: Int,
        draftCommentId: Int,
        sourceOwnerId: Int,
        draftCommentBody: String?
    )

    fun setupReplyViews(replyTo: String?)
    fun replaceBodySelectionTextTo(replyText: String?)
    fun goToCommentEdit(accountId: Int, comment: Comment, commemtId: Int?)
    fun goToWallPost(accountId: Int, postId: Int, postOwnerId: Int)
    fun goToVideoPreview(accountId: Int, videoId: Int, videoOwnerId: Int)
    fun banUser(accountId: Int, groupId: Int, user: User)
    fun displayAuthorAvatar(url: String?)
    fun showAuthorSelectDialog(owners: List<Owner>)
    fun scrollToPosition(position: Int)
    fun showCommentSentToast()
    fun setupOptionMenu(
        topicPollAvailable: Boolean,
        gotoSourceAvailable: Boolean,
        @StringRes gotoSourceText: Int?
    )

    fun setEpmtyTextVisible(visible: Boolean)
    fun setCenterProgressVisible(visible: Boolean)
    fun displayDeepLookingCommentProgress()
    fun dismissDeepLookingCommentProgress()
    fun setCanSendSelectAuthor(can: Boolean)
    fun updateStickers(items: List<Sticker>)
    interface ICommentContextView {
        fun setCanEdit(can: Boolean)
        fun setCanDelete(can: Boolean)
        fun setCanBan(can: Boolean)
    }
}